pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -377,12 +381,39 @@ func (p *Plugin) fetchAliasesPage(ctx context.Context, marker *string) (*string,
 			if err != nil {
 				return nil, err
 			}
-			p.log.Debug("Loaded key", keyIDTag, *alias.TargetKeyId, aliasTag, *alias.AliasName)
+			p.log.Debug("Key loaded", keyIDTag, *alias.TargetKeyId, aliasTag, *alias.AliasName)
 		}
 	}
 	return aliasesResp.NextMarker, nil
 }
 
+func (p *Plugin) scheduleDeleteTask(ctx context.Context) {
+	t := p.clock.Ticker(30 * time.Second)
+	defer t.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			p.log.Debug("Stopping schedule delete task", "reason", ctx.Err())
+			return
+		case <-t.C:
+			keyID, ok := <-p.scheduleDelete
+			if !ok {
+				continue
+			}
+			_, err := p.kmsClient.ScheduleKeyDeletion(ctx, &kms.ScheduleKeyDeletionInput{
+				KeyId:               aws.String(keyID),
+				PendingWindowInDays: aws.Int32(7),
+			})
+			if err != nil {
+				p.log.Error("It was not possible to schedule deletion for key", "reason", err, keyIDTag, keyID)
+				p.scheduleDelete <- keyID
```

Comment:
may we get error code to verify if we really want to reschedule something?

Reply:
Could you elaborate a little on this one? I'm not sure I understand what you mean.

Reply:
aws provides error codes (https://docs.aws.amazon.com/kms/latest/APIReference/API_ScheduleKeyDeletion.html) the ones that worry me are:
InvalidArnException
NotFoundException

those cases may never recover.

Reply:
mmm this might be a good approach to remove the ticker too. Just reading from the channel and dropping the ones that come with those errors.